### PR TITLE
SONARJAVA-2249 ignore classes without static members

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/UtilityClassWithPublicConstructorCheck.java [hunk 1]
```diff
@@ -80,17 +80,22 @@ private static boolean extendsAnotherClassOrImplementsSerializable(ClassTree cla
   }
 
   private static boolean hasOnlyStaticMembers(ClassTree classTree) {
-    if (classTree.members().isEmpty()) {
+    List<Tree> members = classTree.members();
+    if (members.isEmpty() || noStaticMember(members)) {
```
Comment (saberduck) [inline java-checks/src/main/java/org/sonar/java/checks/UtilityClassWithPublicConstructorCheck.java]: `members.isEmpty()` is redundant here, because `noStaticMember` will also be true when there are no classes.
Whole method can be perhaps replaced if logic is cleaned up. We need to test that when there is at least one member, which is not constructor nor empty statement, all of them are static.
```
private static boolean hasOnlyStaticMembers(ClassTree classTree) {
    List<Tree> members = classTree.members().stream()
      .filter(m -> !m.is(Tree.Kind.CONSTRUCTOR, Tree.Kind.EMPTY_STATEMENT))
      .collect(Collectors.toList());
    return !members.isEmpty() && members.stream().allMatch(UtilityClassWithPublicConstructorCheck::isStatic);
  }
```

Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/UtilityClassWithPublicConstructorCheck.java]: I'm not really convinced. the `or` operator is short-circuit, so if members are empty, we won't call `noStaticMember`. I also would like to avoid collecting and creating a new list.

File: java-checks/src/main/java/org/sonar/java/checks/UtilityClassWithPublicConstructorCheck.java [hunk 2]
```diff
@@ -81,15 +81,10 @@ private static boolean extendsAnotherClassOrImplementsSerializable(ClassTree cla
...
+    if (noStaticMember(members)) {
       return false;
     }
-    for (Tree member : members) {
-      if (!isConstructor(member) && !isStatic(member) && !member.is(Tree.Kind.EMPTY_STATEMENT)) {
-        return false;
-      }
-    }
-    return true;
+    return !members.stream().anyMatch(member -> !isConstructor(member) && !isStatic(member) && !member.is(Tree.Kind.EMPTY_STATEMENT));
```
Comment (saberduck) [inline java-checks/src/main/java/org/sonar/java/checks/UtilityClassWithPublicConstructorCheck.java]: Sorry to bother again with this, but this is very negative  :-) How about 
```
return members.stream().allMatch(member -> isConstructor(member) || isStatic(member) || member.is(Tree.Kind.EMPTY_STATEMENT));
```
